python/opendataloader-pdf-mcp/src/opendataloader_pdf_mcp/jobs.py

```
@@ -0,0 +1,192 @@
+                opendataloader_pdf.convert(**kwargs)
+
+                if job._cancel_event.is_set():
+                    job.status = JobStatus.CANCELLED
+                    job.completed_at = _now()
+                    return
+
+                output_file = Path(tmp_dir) / f"{input_file.stem}{ext}"
+                if not output_file.is_file():
+                    matches = sorted(f for f in Path(tmp_dir).iterdir() if f.suffix == ext)
+                    if not matches:
+                        raise RuntimeError(f"No '{ext}' output file generated")
```

Ruff (0.15.12)

[warning] Abstract `raise` to an inner function

(TRY301)